documentation/decisions/infra/0009-environment-use-cases.md

```diff
@@ -0,0 +1,44 @@
+# Determining The Use Cases Of Our Environments
+
+- **Status:** Active
+- **Last Modified:** 2024-01-24
+- **Related Issue:** [#1055](https://github.com/HHS/simpler-grants-gov/issues/1055)
+- **Author:** Kai Siren
+- **Deciders:** Aaron Couch, Alsia Plybeah, James Bursa, Michael Chouinard, Sammy Steiner
+
+## Context and Problem Statement
+
+At present we have two environments: `prod`, and `dev`. This setup is insufficient for a team of our size, so this ADR presents an alternative set of environments. The goal is for these environments meet our needs for years to come.
+
+## Decision Drivers
+
+Not having a dedicated place for spinning up testing environments makes certain kinds of changes impossible to test. You must instead "plan and pray" that the changes apply successfully. This can add hours to days to the amount of time it takes to successfully complete a task. `dev` should function as this environment.
+
+Not having a dedicated environment that's always up to date with `main` (and doesn't have random testing changes) makes it harder for non-engineers to know what to test against. `staging` should function as this environment.
+
+## Desired State
+
+- `prod` - This environment will function identically to its existing behavior, wherein it is always the latest Github release.
+- `staging` - This environment will function the way our current `dev` does, wherein it is always deployed from the `main` branch in Github.
```

Comment:
This doesn't happen right now in staging, is there a plan to start doing that?

Reply:
Yeah, the plan is here: https://github.com/HHS/simpler-grants-gov/issues/1273

Reply:
Love this new approach! 

This is probably out of scope for this ADR, but curious if we plan to set up any QA tests that will run in staging after a deploy as part of the CD process, maybe in #1273?

Reply:
@widal001 I love that idea, but its out of scope for both this ADR and that 10k. We probably want a separate 10k that's exclusively about staging QA testing.

Reply:
Sounds good! Yeah no need to prioritize in the immediate future, but glad this work lays an important foundation for that kind of work.

Love the idea of creating a 10k ticket for it and leaving it in the "tech improvement" backlog until we find a sprint or 30k where we might want to prioritize it. I imagine the decision to use playright for E2E tests #1345 will help with setting up QA tests as well.

Reply:
@widal001 I created a placeholder 10k, but it should get filled out by someone on the frontend eng / product side of the team: https://github.com/HHS/simpler-grants-gov/issues/1369